pkg/admission/crdnooverlappinggvr/crdnooverlappinggvr_admission.go

```diff
@@ -0,0 +1,131 @@
+	"k8s.io/client-go/tools/clusters"
+)
+
+const (
+	PluginName = "apis.kcp.dev/CRDNoOverlappingGVR"
+)
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(_ io.Reader) (admission.Interface, error) {
+			return &crdNoOverlappingGVRAdmission{
+				Handler: admission.NewHandler(admission.Create),
+			}, nil
+		})
+}
+
+type crdNoOverlappingGVRAdmission struct {
+	*admission.Handler
+
+	apiBindingLister        apislisters.APIBindingLister
+	apiExportLister         apislisters.APIExportLister
+	apiResourceSchemaLister apislisters.APIResourceSchemaLister
+}
+
+// Ensure that the required admission interfaces are implemented.
+var _ = admission.ValidationInterface(&crdNoOverlappingGVRAdmission{})
+
+func (p *crdNoOverlappingGVRAdmission) SetKcpInformers(informers kcpinformers.SharedInformerFactory) {
+	p.SetReadyFunc(func() bool {
+		return informers.Apis().V1alpha1().APIBindings().Informer().HasSynced() &&
+			informers.Apis().V1alpha1().APIExports().Informer().HasSynced() &&
+			informers.Apis().V1alpha1().APIResourceSchemas().Informer().HasSynced()
+	})
+	p.apiBindingLister = informers.Apis().V1alpha1().APIBindings().Lister()
+	p.apiExportLister = informers.Apis().V1alpha1().APIExports().Lister()
+	p.apiResourceSchemaLister = informers.Apis().V1alpha1().APIResourceSchemas().Lister()
+}
+
+// Validate checks if the given CRD's Group and Resource don't overlap with other CRDs exposed via APIBindings
+func (p *crdNoOverlappingGVRAdmission) Validate(ctx context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	if a.GetResource().GroupResource() != apiextensions.Resource("customresourcedefinitions") {
+		return nil
+	}
+
+	if a.GetKind().GroupKind() != apiextensions.Kind("CustomResourceDefinition") {
+		return nil
+	}
+	crd, ok := a.GetObject().(*apiextensions.CustomResourceDefinition)
+	if !ok {
+		return fmt.Errorf("unexpected type %T", a.GetObject())
+	}
+	clusterName, err := request.ClusterNameFrom(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster from context: %w", err)
+	}
+
+	apiBindingsForCurrentClusterName, err := p.listAPIBindingsFor(clusterName)
+	if err != nil {
+		return err
+	}
+	for _, apiBindingForCurrentClusterName := range apiBindingsForCurrentClusterName {
+		apiExportClusterName, err := getAPIExportClusterName(apiBindingForCurrentClusterName)
+		if err != nil {
+			// since all exports needs to be checked we need to stop in case of an error
+			return err
+		}
+		apiExport, err := p.getAPIExport(apiExportClusterName, apiBindingForCurrentClusterName.Spec.Reference.Workspace.ExportName)
+		if err != nil {
+			return err
+		}
+
+		for _, apiExportSchemaName := range apiExport.Spec.LatestResourceSchemas {
+			apiResourceSchema, err := p.getAPIResourceSchema(apiExportClusterName, apiExportSchemaName)
+			if err != nil {
+				return err
+			}
+			if apiResourceSchema.Spec.Group == crd.Spec.Group && apiResourceSchema.Spec.Names.Plural == crd.Spec.Names.Plural {
```

Comment:
I think comparing `Group` and `Resource` is enough, we don't have to take `Versions` into consideration.

Reply:
yes, all versions should be owned either by the CRD or the binding.